### DISCUSSION TITLE
[Feedback Wanted 🙏] How Do You Create a New WASDK App?

### DISCUSSION THREAD

**Discussion by karkarl** [microsoft/WinUI-Gallery#1451] [Q&A]
Hey everyone! I'm wanting to understand the WASDK developer journey better, starting with the OOBE (Out Of Box Experience). When you want to start building a Windows App, where do you start?

**Comment by marcelwgn** [comment]
I'm usually starting out with a "WinUI 3 app with packaging project" using the Visual Studio "File > New Project" dialog. If I know I'll add more projects later (e.g. tests), I then move the project into a src folder since I really like [David Fowlers suggested project structure](https://gist.github.com/davidfowl/ed7564297c61fe9ab814). After that I start copying CI/CD files from another projects into the project if I need it (because they were difficult to get right and there isn't much documentation on this) and then also setup a separate package identity for debug builds to prevent name clashes between local version and deployed versions. Once I'm done with that, it's "just" developing the app.

In the past I've also used Windows Template Studio in the past, but it often felt overkill or too opinionated and also didn't follow the project structure I prefer nor did it add unit test projects.

**Comment by ghost1372** [comment]
i have my own Project Template and Wizard, i know that there is a Template Studio, but it is very confusing and heavy, so i created my own Wizard.
First, I choose a template: Blank, MVVM, NavigationView,... Then I can configure options like Pages, CSProj, Libraries, Publishing Stuffs  and more...
I don't need to copy and paste code or files like XamlStyler, Github Actions, EditorConfig, etc. Because my wizard will easily add all of them if you choose.

When the project is created, everything is ready, such as title bar, theme service, settings Page and more... 
Therefore, I can jump right into coding without spending time on the initial setup of the project.

![image](https://github.com/microsoft/WinUI-Gallery/assets/9213496/491bc326-724d-4615-92c0-f301c342edc7)

![image](https://github.com/microsoft/WinUI-Gallery/assets/9213496/b3e3b737-1466-4add-9e2b-81243fd6bd04)

![image](https://github.com/microsoft/WinUI-Gallery/assets/9213496/d0aa5e12-8c1c-4603-8215-1a2d0566af4a)

![image](https://github.com/microsoft/WinUI-Gallery/assets/9213496/844bed59-f6ea-4556-b893-576e0d737834)

![image](https://github.com/microsoft/WinUI-Gallery/assets/9213496/be18d704-fac2-4571-b7b8-551359918cb5)

![image](https://github.com/microsoft/WinUI-Gallery/assets/9213496/2d890b57-8e27-47f0-b230-005a6eb63491)

![image](https://github.com/microsoft/WinUI-Gallery/assets/9213496/6e030aa2-a70a-4b66-95cd-a25fb746deba)
